src/API/HealthController.ts

```diff
@@ -0,0 +1,82 @@
+import { inject, injectable } from 'inversify'
+import { Db, Collection } from 'mongodb'
+import { HealthControllerConfiguration } from './HealthControllerConfiguration'
+@injectable()
+export class HealthController {
+  private readonly db: Db
+  private readonly collection: Collection
+  private readonly configuration: HealthControllerConfiguration
+
+  constructor(
+    @inject('DB') db: Db,
+    @inject('HealthControllerConfiguration') configuration: HealthControllerConfiguration
+  ) {
+    this.db = db
+    this.collection = this.db.collection('health')
+    this.configuration = configuration
+  }
+
+  async getHealth(): Promise<any> {
```

Comment:
We are relying on mongo being up here in order to get any health information. IMO it shouldn't be required for mongo to be working to get a health status of connections to services. services being mongo, ipfs, rabbitmq, blockchain info, wallet info etc.

We will need mongo for health information about specific failures and stuff, but that is extra information.

@lautarodragan not sure if you have anything to add. disagree/agree.

Reply:
Also, I would split up this PR. Tackling everything requested in this issue all in one PR is not necessary.

Reply:
See my response to your other comment on this matter.

I'm thinking now: if MDB is down, there's no way to persist state, so the application should crash. 

Regarding splitting this PR: I'd like your input @kennylavender on how information about retries is being stored in the Health module. @jwicks31 you're storing information inside a single object in the collection, in an array, right? Haven't been able to test it and see how the collection looks in that case yet, but it's probably better to just store aggregated data (just counts, no information about individual IPFS hashes). 

The results of calling `GET /health` should indicate whether there's a problem and trigger alarms, but need not provide debug detail on the problem. In the case Health signals a problem, we'll want to clone production to another environment and manually debug it, most likely. I believe we can have automated replicas without much effort with Atlas (@WesleyCharlesBlake can you confirm that?) 

If we choose to go that path, we could do it in a separate PR and simplify this one for now.

Reply:
@lautarodragan 

> but it's probably better to just store aggregated data (just counts, no information about individual IPFS hashes)

yup, I agree, just counts, no details, just enough info to alert us to an issue to look into.

## Functional Tests
I think we should have functional tests for this endpoint too so we don't accidentally break these health checks. This would probably require collab with @warrenv . This is another good reason to split this PR and focus on getting the functional tests in place as well.

Reply:
It's decided then. Let's do just counts, no details @jwicks31.

Some basic functional tests should be simple enough to set up if you wanna give it a go: 

```ts
// create work etc
await delay(n)
const health = client.getHealth()
// assert predictable changes to health.blocks, etc
```

Reply:
@lautarodragan Yes this is all currently stored inside a single object in the collection, and it is an Array. I'll update it so that we aren't storing any additional information, just the counts of IPFSFailures, IPFSHardFailures, IPFSSoftWIllRetry, and IPFSSoftWontRetry. 

As far as collecting the information from Bitcoin-Core - should we collect this information on an interval and sore the information in a database to be retrieved whenever the /health route is pinged? Or should we just get that information as the /health route is pinged and not store it?